dashboard/test/models/pd/application/application_base_test.rb

```diff
@@ -18,24 +18,33 @@ class ApplicationBaseTest < ActiveSupport::TestCase
           'Application type is not included in the list',
           'Application year is not included in the list',
           'Type is required'
-        ],
-        application.errors.full_messages
+        ].sort,
+        application.errors.full_messages.sort
       )
     end
 
     test 'derived classes override type and year' do
-      application = TEACHER_APPLICATION_CLASS.new
+      application = create TEACHER_APPLICATION_FACTORY
       assert_equal TEACHER_APPLICATION, application.application_type
       assert_equal APPLICATION_CURRENT_YEAR, application.application_year
     end
 
     test 'default status is unreviewed' do
-      application = ApplicationBase.new
+      application = create TEACHER_APPLICATION_FACTORY
```

Comment:
what do you mean by a "whole application" -- do you just mean that you're instantiating the teacher application (a subclass), or do you mean to emphasize that in order to do that you have to fill in all the fields so that it passes validations? I think these are both fine -- the factory seems to take care of filling in those details for you, and I honestly don't see the value in a unit tests which just instantiates and abstract base class, something we'd never do in our application. if there was some specific logic within ApplicationBase that we wanted to test, we could extract that logic into a particular method on ApplicationBase and write a unit test specifically against that method.